enterprise/admin/__init__.py

```diff
@@ -831,22 +833,59 @@ def get_fields(self, request, obj=None):
         return fields
 
 
+class BigTableMysqlPaginator(Paginator):
+    """
+    A paginator that uses INFORMATION_SCHEMA.TABLES to estimate
+    the total number of rows in a table.
+    """
+    ARBITRARILY_LARGE_NUMBER = 10000000
+
+    # pylint: disable=attribute-defined-outside-init
+    @property
+    def count(self):  # pylint: disable=invalid-overridden-method
+        """
+        Returns the number of items in the object list (possibly an estimate).
+        """
+        query = self.object_list.query
+
+        if query.where:
+            return super().count
+        if not getattr(self, '_whole_table_count', None):
+            try:
+                with connection.cursor() as cursor:
+                    cursor.execute(
+                        'SELECT TABLE_ROWS FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = %s',
+                        [query.model._meta.db_table]
+                    )
+                    self._whole_table_count = int(cursor.fetchone()[0])
+            except Exception:  # pylint: disable=broad-except
+                self._whole_table_count = self.ARBITRARILY_LARGE_NUMBER
```

Comment:
nit: Is it worth adding a `logger.exception` here for more visibility into when the raw MySQL query raises an exception for whatever reason? Without it, would the only way we know something went wrong be based on the fact that the table says there are exactly `self.ARBITRARILY_LARGE_NUMBER` number of rows?